src/main/java/com/solvd/api/BaseRestAssuredMethod.java

```diff
@@ -2,84 +2,47 @@
             try {
-                jsonBody = objectMapper.writeValueAsString(requestBody);
+                String jsonBody = objectMapper.writeValueAsString(requestBody);
+                requestSpec.contentType(ContentType.JSON).body(jsonBody);
             } catch (JsonProcessingException e) {
                 throw new IllegalArgumentException("Failed to serialize request body to JSON", e);
             }
-            requestSpec.contentType(ContentType.JSON)
-                    .body(jsonBody);
         }
-
-        return switch (methodType) {
-            case GET -> requestSpec.get(endpointUrl);
-            case POST -> requestSpec.post(endpointUrl);
-            case PUT -> requestSpec.put(endpointUrl);
-            case PATCH -> requestSpec.patch(endpointUrl);
-            case DELETE -> requestSpec.delete(endpointUrl);
-            default -> throw new IllegalArgumentException("Unsupported HTTP method: " + methodType);
-        };
+        return requestSpec;
     }
 
-    public Response execute(HttpMethodTypeEnum methodType) {
-        RequestSpecification requestSpec = RestAssured.given()
-                .header("Authorization", "Bearer " + BEARER_TOKEN)
-                .log().all();
-
-        if (methodType == HttpMethodTypeEnum.POST || methodType == HttpMethodTypeEnum.PUT || methodType == HttpMethodTypeEnum.PATCH) {
-            requestSpec.contentType(ContentType.JSON);
-        }
-
-        return switch (methodType) {
-            case GET -> requestSpec.get(endpointUrl);
-            case POST -> requestSpec.post(endpointUrl);
-            case PUT -> requestSpec.put(endpointUrl);
-            case PATCH -> requestSpec.patch(endpointUrl);
-            case DELETE -> requestSpec.delete(endpointUrl);
-            default -> throw new IllegalArgumentException("Unsupported HTTP method: " + methodType);
-        };
-    }
-
-    public Response executeGraphQL(HttpMethodTypeEnum methodType, GraphQlQuery query) {
-        RequestSpecification requestSpec = RestAssured.given()
-                .header("Authorization", "Bearer " + BEARER_TOKEN)
-                .header("Content-Type", "application/json")
-                .log().all();
-
-        requestSpec.body(query);
-
+    protected Response sendRequest(RequestSpecification spec, HttpMethod methodType) {
         Response response = switch (methodType) {
-            case GET -> requestSpec.get(endpointUrl);
-            case POST -> requestSpec.post(endpointUrl);
-            case PUT -> requestSpec.put(endpointUrl);
-            case PATCH -> requestSpec.patch(endpointUrl);
-            case DELETE -> requestSpec.delete(endpointUrl);
+            case GET -> spec.get(endpointUrl);
+            case POST -> spec.post(endpointUrl);
+            case PUT -> spec.put(endpointUrl);
+            case PATCH -> spec.patch(endpointUrl);
+            case DELETE -> spec.delete(endpointUrl);
```

Comment:
try to use Response response = spec.request(methodType.name(), endpointUrl); instead of switch-case